### PR TITLE
add data download via fdsn for local sensors

### PR DESCRIPTION
**Pull request by HerrMuellerluedenscheid**
data download for local sensors

### CODE REVIEW

**Comment by HerrMuellerluedenscheid** [comment]
depends on #10 
FYI @miili

File: quakesaver_client/fdsnws.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+) -> Path:
+    """Request FDSN waveform dat of the sensor."""
+    logging.debug("requesting waveform data for sensor %s.", uri)
+    location_to_store = assure_output_path(location_to_store)
+    response = requests.get(
+        url=f"{uri}/fdsnws/dataselect/1/query",
+        params=params.json(),
+    )
+
+    if response.status_code != 200:
```
Comment (miili) [inline quakesaver_client/fdsnws.py:51]: Use `raise_for_status()`

File: quakesaver_client/fdsnws.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        raise CorruptedDataError(response.text)
+
+    filename = response.headers["Content-Disposition"].split("=")[1]
```
Comment (miili) [inline quakesaver_client/fdsnws.py:54]: Use `.get` on the dict to be more defensive.

File: quakesaver_client/fdsnws.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    storage_path = location_to_store / filename
+    with open(storage_path, "wb") as file:
+        file.write(response.content)
```
Comment (miili) [inline quakesaver_client/fdsnws.py:57]: Iterate over chunks? The data can be quite big.

File: quakesaver_client/models/local_sensor.py [hunk 4]
```diff
@@ -163,11 +169,19 @@ def get_waveform_stream(self: LocalSensor) -> WebsocketHandler:
     def get_waveform_data(
         self: LocalSensor,
         start_time: datetime,
-        end_time: datetime,
+        end_time: datetime.utcnow(),
```
Comment (miili) [inline quakesaver_client/models/local_sensor.py:172]: Use `datetime.now(tz=timezone.utc)`. `utcnow` does not have TZ information

File: quakesaver_client/fdsnws.py (hunk 3 above)
Comment (HerrMuellerluedenscheid) [reply quakesaver_client/fdsnws.py:57]: I would remove the dump to file in both local and cloud client altogether. But later when there is more time to do so.

File: quakesaver_client/fdsnws.py (hunk 1 above)
Comment (HerrMuellerluedenscheid) [reply quakesaver_client/fdsnws.py:51]: Isn't `raise_for_status` only addressing error codes >400?

File: quakesaver_client/fdsnws.py (hunk 2 above)
Comment (HerrMuellerluedenscheid) [reply quakesaver_client/fdsnws.py]: done

File: quakesaver_client/fdsnws.py (hunk 1 above)
Comment (HerrMuellerluedenscheid) [reply quakesaver_client/fdsnws.py:51]: I'm asking cause the I saw `NoData = Literal[204, 404]  # HTTP Error codes` being used where `204` would slip through, wouldn't it? 🤔

Comment (miili) [reply quakesaver_client/fdsnws.py:51]: Maybe distinguish > 400 and 204?

**Pull request merged**